packages/design-picker/src/utils/index.ts

```diff
@@ -1,3 +1,27 @@
 export * from './available-designs-config';
 export * from './available-designs';
 export * from './fonts';
+import type { Category, Design } from '../types';
+
+export function gatherCategories( designs: Design[] ): Category[] {
+	const allCategories = new Map(
+		designs.flatMap( ( { categories } ) =>
+			categories.map( ( { slug, name } ) => [ slug, name ] as [ string, string ] )
+		)
+	);
+
+	return [ ...allCategories.entries() ].map( ( [ slug, name ] ) => ( { slug, name } ) );
+}
+
+export function filterDesignsByCategory(
+	designs: Design[],
+	selectedCategory: string | null
```

Comment:
I think this one should be a `Category`? Using slug string as a category seems to be confused.

Reply:
Interesting, I thought it'd be more confusing using the category object. Passing the slug around as an object identifier seems like a fairly common pattern I've seen in the code. If I switched `selectedCategory` to an object would that meant the `useState()` should hold a reference to an object instead of to a string id?

Reply:
Maybe it is a bit confusing ... what if I changed the name to `categorySlug`? It is a bit strange talking about "selected" in the context of this util function.

If I did pass a full `Category` type as the thing to search for, I'd still be using the slug for comparison since I can't depend on object comparison.

Reply:
Yeah, I think `categorySlug` is better 👍